graviti/manager/dataset.py

```diff
@@ -5,34 +5,30 @@
 
 """The implementation of the Dataset and DatasetManager."""
 
-from typing import Any, Dict, Generator, Iterator, KeysView, Mapping, Optional
+from typing import Any, Dict, Generator, Iterator, KeysView, Mapping, Optional, Tuple, Type, TypeVar
 
 from tensorbay.dataset import Notes, RemoteData
 from tensorbay.label import Catalog
-from tensorbay.utility import URL
-
-from graviti.client import (
-    get_catalog,
-    get_dataset,
-    get_notes,
-    list_data_details,
-    list_datasets,
-    list_segments,
-)
+from tensorbay.utility import URL, AttrsMixin, attr, common_loads
+
+from graviti.client import get_catalog, get_notes, list_data_details, list_segments
 from graviti.dataframe import DataFrame
 from graviti.exception import ResourceNotExistError
 from graviti.manager.branch import BranchManager
 from graviti.manager.commit import CommitManager
 from graviti.manager.draft import DraftManager
 from graviti.manager.lazy import PagingList
 from graviti.manager.tag import TagManager
+from graviti.openapi import create_dataset, delete_dataset, get_dataset, list_datasets
 from graviti.schema import Extractors, catalog_to_schema, get_extractors
-from graviti.utility import LazyFactory, LazyList, NestedDict
+from graviti.utility import LazyFactory, LazyList, NestedDict, ReprMixin
 
 LazyLists = NestedDict[str, LazyList[Any]]
 
 
-class Dataset(Mapping[str, DataFrame]):  # pylint: disable=too-many-instance-attributes
+class Dataset(  # pylint: disable=too-many-instance-attributes
+    AttrsMixin, ReprMixin, Mapping[str, DataFrame]
```

Comment:
shall we set `_repr_type`?

Reply:
_repr_type = ReprType.INSTANCE
do not need to set here, this is the default value

Reply:
yes the key is sheet name, will it make something wrong ?

Reply:
So when we get a dataset, all segments will be initialized?

Reply:
ReprMixin only effects `__repr__`